jffs2/libjffs2.c

```diff
@@ -496,20 +506,30 @@ static int jffs2_srv_unlink(jffs2_partition_t *p, oid_t *dir, const char *name)
 }
 
 
-static int jffs2_srv_create(jffs2_partition_t *p, oid_t *dir, const char *name, size_t namelen, oid_t *oid, int type, int mode, oid_t *dev)
+static int libjffs2_create(void *info, oid_t *dir, const char *name, oid_t *oid, unsigned mode, int type, oid_t *dev)
 {
 	struct inode *idir, *inode;
 	struct dentry *dentry, *dtemp;
 	int ret = 0;
-	struct jffs2_sb_info *c = JFFS2_SB_INFO(p->sb);
+	struct jffs2_sb_info *c;
+	jffs2_partition_t *p;
+
+	size_t namelen = strlen(name);
+
+
+	if (info == NULL)
+		return -EINVAL;
+
+	p = (jffs2_partition_t *)info;
+	c = JFFS2_SB_INFO(p->sb);
 
 	if (jffs2_is_readonly(c))
 		return -EROFS;
 
-	if (name == NULL || !strlen(name))
+	if (name == NULL || !namelen)
```

Comment:
Here you check `name == NULL` and before that you call `strlen(name)`.

Reply:
Thanks for notice. Done.